File: administrative_duties/events/models/event.py
class Event:
    name: str
    organizer_name: str
    category: str
    description: str
    day_of_week: int
    start_time: str
    duration_time: str
    end_time: str

    def __init__(self, **kwargs):
        self.__dict__.update(kwargs)

    def __getitem__(self, item):
        return getattr(self, item)

File: app/src/main/java/com/ebsolutions/models/Event.java
package com.ebsolutions.models;

import io.micronaut.core.annotation.NonNull;
import io.micronaut.serde.annotation.Serdeable;
import jakarta.validation.constraints.NotBlank;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.SuperBuilder;
import lombok.extern.slf4j.Slf4j;

import java.time.DayOfWeek;
import java.time.LocalTime;

@Data
@Serdeable
@Slf4j
@SuperBuilder
@AllArgsConstructor
@NoArgsConstructor
public class Event extends Base {
    private String eventId;
    @NotBlank
    private String locationId;
    @NotBlank
    private String organizerId;
    @NotBlank
    private String name;
    private String category;
    @NotBlank
    private String description;
    @NonNull
    private DayOfWeek dayOfWeek;
    private LocalTime startTime;
    private short duration;
}
